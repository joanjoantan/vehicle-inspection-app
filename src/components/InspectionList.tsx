import React from "react";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import Paper from "@mui/material/Paper";
import Box from "@mui/material/Box";

import { InspectionListProps } from "./InspectionListInterface";

const InspectionList: React.FC<InspectionListProps> = ({ inspections }) => {
  return (
    <Box className="box-container">
      <h2 className="primary-colour">Available Inspections</h2>
      <TableContainer component={Paper}>
        <Table sx={{ minWidth: 650 }} size="small" aria-label="table">
          <TableHead>
            <TableRow>
              <TableCell>Inspection Name</TableCell>
              <TableCell>Inspection Allowed Types</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {inspections.map((inspection) => (
              <TableRow key={inspection.id}>
                <TableCell component="th" scope="row">
                  {inspection.name}
                </TableCell>
                <TableCell component="th" scope="row">
                  {inspection.allowedTypes.join(", ")}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </Box>
  );
};

export default InspectionList;
