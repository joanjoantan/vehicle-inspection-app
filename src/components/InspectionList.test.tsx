import React from "react";
import { render } from "@testing-library/react";
import InspectionList from "./InspectionList";

// Mock data for inspections
const mockInspections = [
  { id: 1, name: "Inspection 1", allowedTypes: ["car", "van"] },
  { id: 2, name: "Inspection 2", allowedTypes: ["lorry", "any"] },
  // Add more mock inspections as needed
];

test("InspectionList renders correctly", () => {
  const { getByText, getAllByRole } = render(
    <InspectionList inspections={mockInspections} />
  );

  // Check if the title is rendered
  const titleElement = getByText("Available Inspections");
  expect(titleElement).toBeInTheDocument();

  // Check if the table headers are rendered
  const inspectionNameHeader = getByText("Inspection Name");
  const allowedTypesHeader = getByText("Inspection Allowed Types");

  expect(inspectionNameHeader).toBeInTheDocument();
  expect(allowedTypesHeader).toBeInTheDocument();

  // Check if the inspection data is displayed
  const inspection1Name = getByText("Inspection 1");
  const inspection1AllowedTypes = getByText("car, van");
  const inspection2Name = getByText("Inspection 2");
  const inspection2AllowedTypes = getByText("lorry, any");

  expect(inspection1Name).toBeInTheDocument();
  expect(inspection1AllowedTypes).toBeInTheDocument();
  expect(inspection2Name).toBeInTheDocument();
  expect(inspection2AllowedTypes).toBeInTheDocument();

  // Check if there are the correct number of rows in the table body
  const tableRows = getAllByRole("row");
  // Subtract 1 for the table header row
  expect(tableRows.length - 1).toBe(mockInspections.length);
});

test("InspectionList snapshots to compare the rendered component to a saved snapshot", () => {
  const { asFragment } = render(
    <InspectionList inspections={mockInspections} />
  );

  expect(asFragment()).toMatchSnapshot();
});
