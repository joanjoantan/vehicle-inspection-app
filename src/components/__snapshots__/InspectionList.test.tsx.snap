// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InspectionList renders correctly 1`] = `
<DocumentFragment>
  <div
    class="box-container MuiBox-root css-0"
  >
    <h2
      class="primary-colour"
    >
      Available Inspections
    </h2>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="table"
        class="MuiTable-root css-quj9j6-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              Inspection Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              Inspection Allowed Types
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              Inspection 1
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              car, van
            </th>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              Inspection 2
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              lorry, any
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InspectionList snapshots to compare the rendered component to a saved snapshot 1`] = `
<DocumentFragment>
  <div
    class="box-container MuiBox-root css-0"
  >
    <h2
      class="primary-colour"
    >
      Available Inspections
    </h2>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="table"
        class="MuiTable-root css-quj9j6-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              Inspection Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              Inspection Allowed Types
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              Inspection 1
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              car, van
            </th>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              Inspection 2
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              scope="row"
            >
              lorry, any
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
